### PR TITLE
bugfix/16920-data-module-columntypes

### LINKED ISSUE #16920: Data module removing 0's in fips codes
#### Expected behaviour
The data module does not remove the first 0 in the fips code and the map renders correctly

#### Actual behaviour
The data module removes the first 0 in the fips code and the map does not render any of its points.

An example of a fips code in the Alabama map: 01007

The data module seems to be removing the first 0 in the codes, and so Highmaps cant find the right code to link to.

If you log mapKey and mapMap in the line I've linked below, you'll see that the codes do not match. I assume the data module is doing some custom parsing before it gets to this point and the 0 is getting removed because of it.

https://github.com/highcharts/highcharts/blob/master/ts/Series/Map/MapSeries.ts#L1057

#### Live demo with steps to reproduce
https://jsfiddle.net/vf2yeL5g/

#### Product version
Latest

### PR DESCRIPTION
**Pull request by hubertkozik**
Fixed #16920, formalized and fixed the `data.columnTypes` option.

---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1205897590371460

### CODE REVIEW

**Review by KacperMadej — CHANGES_REQUESTED** [review]
The column-types-map demo reports an error in Utils that is now tracked [here](https://github.com/highcharts/map-collection-dist/issues/134).

File: samples/unit-tests/data/general/demo.js [hunk 1]
```diff
@@ -240,6 +240,20 @@ QUnit.test('Combination charts and column mapping', function (assert) {
         `Name should be mapped correctly from CSV for packed bubble series,
         which is non-cartesian (#19143).`
     );
+
+    chart = Highcharts.chart('container', {
+        data: {
+            csv: 'code;value\n01007;58\n01003;10\n01041;32\n01085;84\n01133;12',
+            columnTypes: ['string', 'float']
```
Comment (KacperMadej) [inline samples/unit-tests/data/general/demo.js:247]: Is `'float'` supported? or should this be `'number'`?

**Review by KacperMadej — APPROVED** [review]
One test related comment below, but other than that LGTM 👍

File: samples/unit-tests/data/general/demo.js (hunk 1 above)
Comment (hubertkozik) [reply samples/unit-tests/data/general/demo.js:247]: `float` is working fine, but I think we should support `number` only.

File: ts/Extensions/Data.ts [hunk 2]
```diff
@@ -2450,6 +2457,22 @@ export default Data;
  * @apioption data.columns
  */
 
+/**
+ * An array option that specifies the data type for each column in the series
+ * loaded within the data module.
+ *
+ * Possible values: `"string"`, `"number"`, `"date"`.
```
Comment (TorsteinHonsi) [inline ts/Extensions/Data.ts:2464]: To help our future Editor (and other GUI generators), we should add:

* A stricter [type](https://github.com/highcharts/highcharts/blob/e6cec31e422a680d9e2a6cc22b10f321d0c90d98/ts/Extensions/Data.ts#L87): `columnTypes?: Array<'string'|'number'|'date'>;
* A `@validvalue` doclet tag, even though this one will soon be redundant as we create the TS-driven API.

**Review by TorsteinHonsi — CHANGES_REQUESTED** [review]
Thanks!

**Review by TorsteinHonsi — APPROVED** [review]

**Pull request merged**